torchrecipes/paved_path/charnn/model.py

```diff
@@ -177,17 +175,13 @@ def _init_weights(self, module) -> None:
             module.bias.data.zero_()
             module.weight.data.fill_(1.0)
 
-    def forward(self, idx, targets=None) -> Tuple[Tensor, Tensor]:
-        b, t = idx.size()
+    def forward(self, idx: Tensor) -> Tensor:
+        _, t = idx.size()
         assert t <= self.block_size, "Cannot forward, model block size is exhausted."
 
         # forward the GPT model
         x = self.emb_stem(idx)
         x = self.blocks(x)
         x = self.ln_f(x)
         logits = self.head(x)
-        loss = None
-        if targets is not None:
-            loss = F.cross_entropy(logits.view(-1, logits.size(-1)), targets.view(-1))
```

Comment:
make sense to move loss out of the main module!

Reply:
This is also required to make model torchscriptable.